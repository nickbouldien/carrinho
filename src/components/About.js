import React from 'react';
import { Container, List } from 'semantic-ui-react';

const About = () => (
  <div id='about-div'>
    <Container>
      <h1>About this project:</h1>

      <List link>
        <List.Item href="https://github.com/nickbouldien/carrinho">github link</List.Item>
      </List>

      <h3>Tech used:</h3>
      <List link>
        <List.Item href="https://github.com/facebook/react">React 16</List.Item>
        <List.Item href='https://redux.js.org/'>redux </List.Item>
        <List.Item href='https://redux-saga.js.org/'>redux-saga</List.Item>
        <List.Item href='https://react.semantic-ui.com/introduction'>Semantic UI React</List.Item>
        <List.Item href='https://redux-saga.js.org/'>redux-saga</List.Item>
        <List.Item href='http://recharts.org/#/en-US'>Recharts</List.Item>
        <List.Item href="https://www.heroku.com/">hosted on Heroku</List.Item>
        <List.Item href='https://firebase.google.com/docs/reference/rest/database/'>Firebase rest api</List.Item>
      </List>

      <h3>Sources:</h3>
      <List link>
        <List.Item href="https://github.com/reactjs/react-redux">react-redux</List.Item>
        <List.Item href='https://redux.js.org/'>redux</List.Item>
      </List>

    </Container>
  </div>
);


export default About;
