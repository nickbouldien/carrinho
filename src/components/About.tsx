import React from 'react';
import {Container, List} from 'semantic-ui-react';
import {Helmet} from 'react-helmet';

const About = () => (
  <Container id="about-div">
    <Helmet>
      <title>Carrinho - about</title>
    </Helmet>

    <h1>About this project:</h1>

    <List link>
      <List.Item href="https://github.com/nickbouldien/carrinho" target="_blank">
        github link
      </List.Item>
    </List>

    <h3>Tech used:</h3>
    <List link>
      <List.Item href="https://github.com/facebook/react" target="_blank">
        React 16
      </List.Item>
      <List.Item href="https://redux.js.org/" target="_blank">
        redux{' '}
      </List.Item>
      <List.Item href="https://redux-saga.js.org/" target="_blank">
        redux-saga
      </List.Item>
      <List.Item href="https://react.semantic-ui.com/introduction" target="_blank">
        Semantic UI React
      </List.Item>
      <List.Item href="https://redux-saga.js.org/" target="_blank">
        redux-saga
      </List.Item>
      <List.Item href="http://recharts.org/#/en-US" target="_blank">
        Recharts
      </List.Item>
      <List.Item href="https://www.heroku.com/" target="_blank">
        hosted on Heroku
      </List.Item>
      <List.Item href="https://firebase.google.com/docs/reference/rest/database/" target="_blank">
        Firebase rest api
      </List.Item>
    </List>

    <h3>Sources:</h3>
    <List link>
      <List.Item href="https://github.com/reactjs/react-redux" target="_blank">
        react-redux
      </List.Item>
      <List.Item href="https://redux.js.org/" target="_blank">
        redux
      </List.Item>
    </List>
  </Container>
);

export default About;
