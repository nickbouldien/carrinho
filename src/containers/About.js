import React from 'react';

const About = () => (
  <div id='about-div'>
    <h1>About this project:</h1>

    <h3>Creator:</h3>
    <p>Nick Bouldien <a href="https://github.com/nickbouldien">github profile</a></p>


    <h3>Tech used:</h3>
    <div>
      <p>- <a href='https://github.com/facebook/react'>React 16</a></p>
      <p>- <a href='https://redux.js.org/'>redux </a></p>
      <p>- <a href='https://redux-saga.js.org/'>redux-sagas </a></p>
      <p>- <a href='https://react.semantic-ui.com/introduction'>Semantic UI React</a></p>
      <p>- <a href='http://recharts.org/#/en-US'>Recharts</a></p>
      
      {/* styled components */}
      {/* <p>- <a href='https://webpack.github.io/'>Webpack 3</a> (too much fun to setup...)</p> */}

    </div>

    <br />

    <div>
      <p>
        API usage:
        <br />
        <a href='https://firebase.google.com/docs/reference/rest/database/'>Firebase rest api</a>

        {/* <p>- <a href='https://github.com/toddmotto/public-apis#currency-exchange'>github link</a></p> */}

      </p>
      
    </div>

    <div>
      Sources:
      <a href="https://github.com/reactjs/react-redux" target="_blank" rel="noopener noreferrer">react-redux</a>
      <a href="https://redux.js.org/" target="_blank" rel="noopener noreferrer">redux</a>


      {/* <a href="https://github.com/redux-saga/redux-saga" target="_blank">redux-saga</a> */}
      {/* thunks (in firebase branch) */}
    </div>

  </div>
);


export default About;
