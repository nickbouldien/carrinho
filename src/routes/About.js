import React from 'react';

const About = () => (
  <div id='about-div'>
    <h1>About this project:</h1>
    <h3>Tech used:</h3>
    <div>
      <p>- <a href='https://github.com/facebook/react'>React 16</a></p>
      {/* <p>- <a href='https://webpack.github.io/'>Webpack 3</a> (too much fun to setup...)</p> */}
      {/* <p>- <a href='https://react-bootstrap.github.io/'>React Bootstrap</a> (quick styling)</p> */}

    </div>

    <div>
      API usage:
      {/* <p>- <a href='https://iextrading.com/api-exhibit-a'>IEX Trading (api used for stocks)</a></p>
      <p>- <a href='http://fixer.io/'>fixer.io (api used for currencies)</a></p> */}

      For list of public apis:
      {/* <p>- <a href='https://github.com/toddmotto/public-apis#currency-exchange'>github link</a></p> */}

    </div>
  </div>
);


export default About;
