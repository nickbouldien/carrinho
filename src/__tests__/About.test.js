import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'react-testing-library';
import About from '../components/About';

describe('test About component', () => {
  const container = document.createElement('div');
  ReactDOM.render(<About />, container);

  test("h1 shows up", () => {
    const h1 = container.querySelector('h1');
    expect(h1.textContent).toMatch("About this project:");
  });

  test("there are 2 h3s", () => {
    const h3s = container.querySelectorAll('h3');
    expect(h3s).toHaveLength(2);
    expect(h3s[0].textContent).toMatch("Tech used:");
  });

  test("there are 11 links", () => {
    const a = container.querySelectorAll('a');
    expect(a).toHaveLength(11);
  });

  test("snapshot test", () => {
    const { container } = render(<About />);
    expect(container.firstChild).toMatchSnapshot();
  });

});
