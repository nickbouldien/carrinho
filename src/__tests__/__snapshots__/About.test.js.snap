// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test About component snapshot test 1`] = `
<div
  id="about-div"
>
  <div
    class="ui container"
  >
    <h1>
      About this project:
    </h1>
    <div
      role="list"
      class="ui link list"
    >
      <a
        href="https://github.com/nickbouldien/carrinho"
        role="listitem"
        class="item"
      >
        github link
      </a>
    </div>
    <h3>
      Tech used:
    </h3>
    <div
      role="list"
      class="ui link list"
    >
      <a
        href="https://github.com/facebook/react"
        role="listitem"
        class="item"
      >
        React 16
      </a>
      <a
        href="https://redux.js.org/"
        role="listitem"
        class="item"
      >
        redux 
      </a>
      <a
        href="https://redux-saga.js.org/"
        role="listitem"
        class="item"
      >
        redux-saga
      </a>
      <a
        href="https://react.semantic-ui.com/introduction"
        role="listitem"
        class="item"
      >
        Semantic UI React
      </a>
      <a
        href="https://redux-saga.js.org/"
        role="listitem"
        class="item"
      >
        redux-saga
      </a>
      <a
        href="http://recharts.org/#/en-US"
        role="listitem"
        class="item"
      >
        Recharts
      </a>
      <a
        href="https://www.heroku.com/"
        role="listitem"
        class="item"
      >
        hosted on Heroku
      </a>
      <a
        href="https://firebase.google.com/docs/reference/rest/database/"
        role="listitem"
        class="item"
      >
        Firebase rest api
      </a>
    </div>
    <h3>
      Sources:
    </h3>
    <div
      role="list"
      class="ui link list"
    >
      <a
        href="https://github.com/reactjs/react-redux"
        role="listitem"
        class="item"
      >
        react-redux
      </a>
      <a
        href="https://redux.js.org/"
        role="listitem"
        class="item"
      >
        redux
      </a>
    </div>
  </div>
</div>
`;
